### PR TITLE
Add command `piv authsign`

### PR DESCRIPTION
**Pull request by jmichelp**
The command allows to select the signature algorithm as well as the key slot to be used for signing.
Only tested on ECC-P256 (I don't have RSA samples at hand)

### CODE REVIEW

**Comment by iceman1001** [comment]
Awesome!

File: client/src/cmdpiv.c [hunk 1]
```diff
@@ -653,6 +653,44 @@
     return PivGetDataByCidAndPrint(channel, &(PIV_CONTAINERS[idx]), decodeTLV, verbose);
 }
 
+static int PivAuthenticateSign(Iso7816CommandChannel channel, uint8_t alg_id, uint8_t key_id, uint8_t nonce[], size_t nonce_len, void **result, bool decodeTLV, bool verbose) {
+    if (nonce_len > 0x7A) {
+        if (verbose == true) {
+            PrintAndLogEx(WARNING, "Nonce cannot exceed %d bytes. Got %d bytes.", 0x7a, nonce_len);
```
[the remark this answers is not in the document]
Comment (jmichelp) [reply client/src/cmdpiv.c]: resolved

File: client/src/cmdpiv.c [hunk 2]
```diff
@@ -653,6 +653,45 @@
     return PivGetDataByCidAndPrint(channel, &(PIV_CONTAINERS[idx]), decodeTLV, verbose);
 }
 
+static int PivAuthenticateSign(Iso7816CommandChannel channel, uint8_t alg_id, uint8_t key_id, uint8_t nonce[], size_t nonce_len, void **result, bool decodeTLV, bool verbose) {
+    const size_t MAX_NONCE_LEN = 0x7a;
+    if (nonce_len > MAX_NONCE_LEN) {
+        if (verbose == true) {
+            PrintAndLogEx(WARNING, "Nonce cannot exceed %d bytes. Got %d bytes.", MAX_NONCE_LEN, nonce_len);
```
[the remark this answers is not in the document]
Comment (jmichelp) [reply client/src/cmdpiv.c]: resolved

[the remark this answers is not in the document]
Comment (jmichelp) [reply client/src/cmdpiv.c]: resolved

**Pull request merged**